python/tests/TestCUDAGraphs.py

```diff
@@ -0,0 +1,59 @@
+__author__ = "Raul P. Pelaez"
+import openmmtorch as ot
+import torch
+import openmm as mm
+import numpy as np
+import pytest
+
+class UngraphableModule(torch.nn.Module):
+    def forward(self, positions):
+        torch.cuda.synchronize()
+        return (0.5*torch.sum(positions**2), -2.0*positions)
+
+class GraphableModule(torch.nn.Module):
+    def forward(self, positions):
+        energy=0.5*torch.einsum('ij,ij->i', positions, positions).sum()
+        return (energy, -2.0*positions)
+
+class GraphableModuleOnlyEnergy(torch.nn.Module):
+    def forward(self, positions):
+        energy=0.5*torch.einsum('ij,ij->i', positions, positions).sum()
+        return (energy)
```

Comment:
These two models yield the same forces and energies when I query OpenMM for them in the tests below. In the first case I am computing the forces manually, but in the second case they are computed by a backward pass on the energy.
I really do not understand why I need to provide energies as 0.5*positions^2 to obtain forces equal to -2*positions. The 0.5 should not be there AFAIK.

Reply:
This is a bug in my implementation and I am investigating.

Reply:
I forgot to reset the position gradients after capturing the graph, which in translates into the force being doubled only during the first call to execute. I fixed that in a commit.